lib/karafka/connection/listener.rb

```diff
@@ -31,27 +27,24 @@ def initialize(consumer_group)
       #   Kafka connections / Internet connection issues / Etc. Business logic problems should not
       #   propagate this far
       def fetch_loop(block)
-        messages_consumer.fetch_loop do |raw_messages|
+        consumer.fetch_loop do |raw_messages|
           block.call(consumer_group.id, raw_messages)
         end
         # This is on purpose - see the notes for this method
         # rubocop:disable RescueException
       rescue Exception => e
         # rubocop:enable RescueException
         Karafka.monitor.notice_error(self.class, e)
-        @messages_consumer&.stop
-        retry if @messages_consumer
+        @consumer&.stop
```

Comment:
How often will this be `nil`?

Reply:
Only if someone sends the shutdown signal before initialized (happened to me once) and in testing. But one is more than 0, that's why it go tthere in the first place.